src/BowlingGame.js

```diff
@@ -0,0 +1,33 @@
+function Bowling(frame = new Frame()) {
+
+  this._currentFrame = frame;
+  this._result = 0;
+  this._frames = [];
+
+};
+
+Bowling.prototype.roll = function(pins) {
+  this._currentFrame.roll(pins);
+
+  if (this._frames.length !== 0) {
+
+    if (this._frames[this._frames.length - 1].isStrike === true) {
+      this._result += pins;
+    }
+
+  };
+
+  this._result += pins;
+
+  if (this._currentFrame.isOver() && this._frames.length < 10){
+
+    this._frames.push(this._currentFrame);
+    this._currentFrame = new Frame();
+
+  };
+};
+
```

Comment:
If you have the time, maybe refactor this section as I am finding it hard to tell what's happening. Possibly focus on your naming conventions (ie Bowling) as it's not completely descriptive of what this object is responsible for. It might also be handy to clean up the layout if possible as I'm unsure which code belongs to what function?